File: 2021년/9월/0923 플로이드.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int n,m;
int dist[101][101];
int INF = 10000001;
int main()
{
	scanf("%d", &n);
	scanf("%d", &m);
	
	for(int i=1;i<n+1;i++)
	{
		for(int j=1;j<n+1;j++)
		{
			if (i==j) dist[i][j] = 0;
			else dist[i][j] = INF; 
		}
	}
	
	for(int i=0;i<m;i++)
	{
		int a,b,c;
		scanf("%d %d %d",&a, &b, &c);
		dist[a][b] = min(dist[a][b], c);
	}
	
	for(int k=1;k<n+1;k++)
	{
		for(int i=1;i<n+1;i++)
		{
			for(int j=1;j<n+1;j++)
			{
				dist[i][j] = min(dist[i][j], dist[k][j]+dist[i][k]);
			}
		}
	}
	
	for(int i=1;i<n+1;i++)
	{
		for(int j=1;j<n+1;j++)
		{
			if(dist[i][j] == INF) printf("%d ", 0);
			else printf("%d ", dist[i][j]);
		}
		printf("\n");
	}
}

File: 2021년/9월/0923 플로이드.py
# 백준_11404_플로이드_플로이드-와샬_골드 4

import sys
input = sys.stdin.readline
INF = sys.maxsize

n = int(input())
m = int(input())

dist = [[INF for _ in range(n)] for _ in range(n)]

for _ in range(m):
    a, b, c = map(int, input().split())
    dist[a-1][b-1] = min(dist[a-1][b-1], c)

for k in range(n):
    for i in range(n):
        for j in range(n):
            dist[i][j] = min(dist[i][j], dist[k][j] + dist[i][k])

for i in range(n):
    for j in range(n):
        if dist[i][j] == INF or i == j:
            print(0, end=' ')
        else:
            print(dist[i][j], end=' ')
    print()
